sklearn/cluster/k_means_.py

```diff
@@ -280,7 +283,14 @@ def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
         raise ValueError('Number of iterations should be a positive number,'
                          ' got %d instead' % max_iter)
 
-    X = as_float_array(X, copy=copy_x)
+    # avoid forcing order when copy_x=False
+    order = "C" if copy_x else None
```

Comment:
The docstring of `copy_x` should be updated both in the estimator class and in the function to explain that setting `copy_x` to `False` will not ensure that the data is C-contiguous possibly causing a significant slowdown.